### ISSUE TITLE
“Test Request” feature doesn’t use selected server

### ISSUE THREAD

**Issue by jpjpjp** [scalar/scalar#3078]
### What happens?

Hi.  I'm having an issue with multiple server support with the "Test Request" feature.   When the spec defines multiple servers, and I select the second one in the drop down, the correct server is populated in the documentation for all the endpoints, but when I hit the "Test Request" feature, the pop-up dialog still has the first server listed.

This discord message https://discord.com/channels/1135330207960678410/1135330208627564597/1282691804747010128 has 3 pictures hopefully more useful than my words

### What did you expect to happen?

The request is sent to the selected server instead of the first one on the list. 

### OpenAPI Document

https://lm-v2-api-v2-proxy-7d8e75f9cd43.herokuapp.com/v2/openapi

**Comment by hanspagel** [comment]
True! I think there’s a fix hidden in one of the open PRs. But if we can get a dedicated PR from @hwkr that would be great, too!

**Comment by hwkr** [comment]
> True! I think there’s a fix hidden in one of the open PRs. But if we can get a dedicated PR from @hwkr that would be great, too!

It sounds like it's going to be in the big PR that @amritk is working on so I'll just leave it with that to avoid conflicts and check that it's fixed after 🙏

**Comment by jpjpjp** [comment]
@hwkr will you close this after that PR is merged, so I know when to grab an update?   If there's another issue I should follow instead just let me know.

**Comment by hwkr** [comment]
> @hwkr will you close this after that PR is merged, so I know when to grab an update? If there's another issue I should follow instead just let me know.

I'll update this ticket when it's merged! The fix should be in #2931

**Comment by jpjpjp** [comment]
@marclave , @hwkr.  Yeah! Looking forward to checking this out.   I read through the PR, but it wasn't obvious to me if there is a version of express-api-reference with this PR yet.  Is there a way to correlate a PR to a release at all?

**Comment by jpjpjp** [comment]
@marclave @hwkr I noticed that a new version of the express-api-reference module was published today so I gave it a try.

In my generated code the "Test Request" feature is still defaulting to the first server in the list, even if the user selected the second one.   One improvement in this release is that from each invoked "Test Request" attempt you now select the server list in the popup and choose one other than the first one, however, I'd suggest that if I set one other than the first one as the default from the main page, that selection should be "remembered".

Screenshots from docs generated by express-api-reference v0.4.156 attached
<img width="653" alt="Screenshot 2024-09-21 at 4 51 45 PM" src="https://github.com/user-attachments/assets/f4fac2a9-b9a8-426e-b3a1-103cd93910cf">
<img width="680" alt="Screenshot 2024-09-21 at 4 49 56 PM" src="https://github.com/user-attachments/assets/633e398a-f650-4475-b896-be48fd6697b5">

**Comment by hwkr** [comment]
@jpjpjp Ahhhh I see, yeah I should be able to have a look

**Comment by amritk** [comment]
@hwkr this one _should_ be working, at least it does on regular references
[screen-recorder-wed-sep-25-2024-09-25-21.webm](https://github.com/user-attachments/assets/3ec7c4c5-cc4d-49f2-b9c8-a4f7e2180a7e)

**Comment by hwkr** [comment]
@jpjpjp I gave it a test with our local express environment and it seems to be working. Would you be able to share your project / repo or a minimal reproduction if you're still having the issue?


https://github.com/user-attachments/assets/a52f5bad-14ad-4e63-9b92-5f0ebe2635a4

**Comment by jpjpjp** [comment]
Hmm. I was surprised to see this working for me now, too.  I may understand what is happening.  It seems that Scalar is remembering the last set value for the server, regardless of if it is set at the top of the page or in a recent "Test Request" dialog.   

Put another way the scope of changing the server in a Test Request is global (it affects all other requests), even though it doesn't modify what appears to be set at the top of the page.  In the attached screen recording I'm trying to demonstrate that while the Server list at the top of the doc is set to the first server, setting it to the second server in the "Test Request" for one endpoint seems to put it to the second server for the "Test Request" for other endpoints, even though it appears that the default (at the top) is still the first server.   I hope that makes sense.

https://github.com/user-attachments/assets/c0de1471-4f9c-47e4-b3f1-c889d2cf9e36

**Comment by jpjpjp** [comment]
By the way, I may have stumbled across another "scope" related issue.   While setting up for that video, I was in a state where I had not selected my security scheme.   I went to make a Test Request for the first endpoint, and then set the scheme to Bearer Auth security (one of two methods supported), and added a value for the auth header, in the Test Request dialog.

After finishing that Test Request, I then went back to the top to try to set the Security Scheme and Token "globally", however, when I try a Test Request at any other endpoint it still shows no Security Selected.  I realize this is a different issue (if you agree its an issue), but thought I'd comment here since they seem related from the context of "scope".

https://github.com/user-attachments/assets/8dbe8e55-5069-40ab-9792-37cb06075703
